disperser/batcher/txn_manager.go

```diff
@@ -251,10 +287,37 @@ func (t *txnManager) monitorTransaction(ctx context.Context, req *TxnRequest) (*
 	var err error
 
 	rpcCallAttempt := func() error {
-		ctxWithTimeout, cancel := context.WithTimeout(ctx, t.txnRefreshInterval)
-		defer cancel()
-		t.logger.Debug("monitoring transaction", "component", "TxnManager", "method", "monitorTransaction", "txHash", req.Tx.Hash().Hex(), "tag", req.Tag, "nonce", req.Tx.Nonce())
+		t.logger.Debug("monitoring transaction", "txHash", req.Tx.Hash().Hex(), "tag", req.Tag, "nonce", req.Tx.Nonce())
+
+		ctxWithTimeout, cancelBroadcastTimeout := context.WithTimeout(ctx, t.txnBroadcastTimeout)
+		defer cancelBroadcastTimeout()
+
+		// Ensure transactions are broadcasted to the network before querying the receipt.
+		// This is to avoid querying the receipt of a transaction that hasn't been broadcasted yet.
+		// For example, when Fireblocks wallet is used, there may be delays in broadcasting the transaction due to latency from cosigning and MPC operations.
+		err = t.ensureAnyTransactionBroadcasted(ctxWithTimeout, req.txAttempts)
+		if err != nil && errors.Is(err, context.DeadlineExceeded) {
```

Comment:
It looks like the error is unhandled if it isn't equal to `context.DeadlineExceeded`?

Reply:
Yeah I think this is the only error that can be returned, but agree it should handle other errors too